websites/M/Messenger/presence.ts

```diff
@@ -2,58 +2,52 @@ const presence = new Presence({
+    },
+    [messageRecipient, callRecipient] = await Promise.all([
+      presence.getSetting<boolean>("message"),
+      presence.getSetting<boolean>("call")
+    ]);
 
-  if (document.location.pathname.includes("/videocall/")) {
-    presenceData.largeImageKey = "messenger";
-
-    user = document.querySelector(
-      "#u_0_0 > div.r30xiam5.m0q0jmkx.alrytcbg.hp5uecnq.g2121wdl > div > div:nth-child(5) > div > div > div > div > div.prklkq8o.t7elcel3.sd0tyowg.ocjcko58.p3f4w9ai.f5zavhip.foed1vyy > div > div > div.ocjcko58.foed1vyy > div > p"
-    );
-    if (!user || !(user as HTMLElement).textContent) {
-      //presenceData.details = "In a video call or";
-      user = "user not found.";
-      presenceData.details = "In videocall with someone";
-      presenceData.smallImageKey = "videocall";
-    } else {
-      //presenceData.details = "In call with:";
-      user = (user as HTMLElement).textContent;
-      presenceData.details = "In call with someone";
+  if (document.location.pathname.includes("/groupcall/")) {
+    const users = document
+      .querySelector(".paneContent h1")
+      ?.textContent?.split(", ")
+      ?.slice(1);
+    if (users) {
+      if (new URLSearchParams(location.search).get("has_video") === "true") {
+        presenceData.details = "In a video call with:";
+        presenceData.smallImageKey = "videocall";
+      } else {
+        presenceData.details = "In a call with:";
+        presenceData.smallImageKey = "call";
+      }
+      presenceData.state = callRecipient ? users.join(", ") : "(Hidden)";
+    } else if (document.querySelector("span")?.textContent) {
+      presenceData.details = "In a Messenger room";
       presenceData.smallImageKey = "call";
-    }
-    //presenceData.state = user;
-    presenceData.state = "(Hidden until presence settings.)"; // Add setting for this when presence settings are a thing!!
+    } else presenceData.details = "Joining a call...";
   } else if (document.location.pathname.includes("/t/")) {
-    presenceData.largeImageKey = "messenger";
-
-    user = document.querySelector("._3oh-");
-    typing = document.querySelector(
-      "body > div > div > div > div:nth-child(2) > span > div._20bp > div._4_j4 > div._4rv3._7og6 > div > div._7kpk > div > div > div:nth-child(1) > div > div > div > div > div > div > span > span"
-    );
-    if (typing === null) {
+    if (!document.querySelector('[data-text="true"]')?.textContent) {
       presenceData.details = "Reading messages from:";
       presenceData.smallImageKey = "reading";
     } else {
       presenceData.details = "Writing to:";
       presenceData.smallImageKey = "writing";
     }
-    //presenceData.state = user.textContent;
-    presenceData.state = "(Hidden until presence settings.)"; // Add setting for this when presence settings are a thing!!
+    presenceData.state = messageRecipient
+      ? document.querySelector("div.t6p9ggj4.tkr6xdv7 span > span")
+          ?.textContent ?? "Unknown"
+      : "(Hidden)";
```

Comment:
pref: Use an if statement here instead and do not include the `(Hidden)` part

Reply:
Is it a requirement? Since if I omit the hidden part I would have to drag `.details` out on several more lines just to make it fit context (at least with my solution)
![image](https://user-images.githubusercontent.com/77577746/147711844-1e4650bd-487d-4239-bf33-9e915552f598.png)
![image](https://user-images.githubusercontent.com/77577746/147711968-47928257-15b1-4e5b-a276-aaad571b1878.png)

Reply:
It is not a requirement, but I think it would look a bit nicer. You could set `details` accordingly too

Your call I guess

Reply:
It looks okay now I guess, and I'll be down in a few minutes and won't be able to write so it's fine to leave it so ig.